homeassistant/components/sensor/google_travel_time.py

```diff
@@ -186,6 +216,53 @@ def update(self):
         elif atime is not None:
             options_copy['arrival_time'] = atime
 
-        self._matrix = self._client.distance_matrix(self._origin,
-                                                    self._destination,
-                                                    **options_copy)
+        # Convert device_trackers to google friendly location
+        if hasattr(self, '_origin_entity_id'):
+            self._origin = self._get_location_from_entity(
+                self._origin_entity_id
+            )
+
+        if hasattr(self, '_destination_entity_id'):
+            self._destination = self._get_location_from_entity(
+                self._destination_entity_id
+            )
+
+        if self._destination is not None and self._origin is not None:
+            self._matrix = self._client.distance_matrix(self._origin,
```

Comment:
@bah2830 Just thinking out loud. In your quest to make google-travel-time a first class HASS citizen, would template language for source and destination not give the `sensor` domain already? And some added goodness

The device_tracker.dev shorthand (entity with has_location) is still very handy though and suggest we keep that

Would probably still need to search for friendly_name, except if @balloob has some template magic for it?